python/baseline/pytorch/transformer.py

```diff
@@ -120,54 +120,29 @@ def forward(self, query, key, value, mask=None):
         return self.w_O(x)
 
 
-class FFN(nn.Module):
-    """
-    FFN from https://arxiv.org/abs/1706.03762 via http://nlp.seas.harvard.edu/2018/04/03/attention.html
-
-    The `FFN` layer is block in the Transformer that follows multi-headed self-attention.  It consists
-    of an expansion from `d_model` to `d_ff` (with sub-sequent relu and dropout), followed by a squeeze
-    layer that pushes it back to `d_model`.  In the `tensor2tensor` codebase, this is implemented as convolution of
-    size 1 over the temporal sequence, which is equivalent, but in PyTorch, we dont need to do anything explicitly,
-    thanks to https://github.com/pytorch/pytorch/pull/1935!
-
-    """
-    def __init__(self, d_model, pdrop, activation_type='relu', d_ff=None):
-        """Constructor, takes in model size (which is the external currency of each block) and the feed-forward size
-
-        :param d_model: The model size.  This is the size passed through each block
-        :param d_ff: The feed-forward internal size, which is typical 4x larger, used internally
-        :param pdrop: The probability of dropping output
-        """
-        super(FFN, self).__init__()
-        if d_ff is None:
-            d_ff = 4 * d_model
-        self.expansion = pytorch_linear(d_model, d_ff)
-        self.squeeze = pytorch_linear(d_ff, d_model)
-        self.dropout = nn.Dropout(pdrop)
-        self.act = pytorch_activation(activation_type)
-
-    def forward(self, x):
-        """Expand to `d_ff` then activation, followed by a squeeze operation back down to `d_model`
-
-        :param x: The output of the previous attention module
-        :return: An output the same size as the input, but transformed
-        """
-        return self.squeeze(self.dropout(self.act(self.expansion(x))))
-
-
 class TransformerEncoder(nn.Module):
     def __init__(self, num_heads, d_model, pdrop, scale=True, activation_type='relu', d_ff=None):
         super(TransformerEncoder, self).__init__()
         self.d_model = d_model
+
```

Comment:
I am getting the following error when trying to make a TransformerEncoder with `d_ff=None`

```traceback
>>> from baseline.pytorch.transformer import TransformerEncoder
>>> te = TransformerEncoder(num_heads=4, d_model=512, pdrop=0.1)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    te = TransformerEncoder(num_heads=4, d_model=512, pdrop=0.1)
  File "/home/blester/dev/work/baseline/python/baseline/pytorch/transformer.py", line 129, in __init__
    self.ffn = nn.Sequential(pytorch_linear(d_model, d_ff),
  File "/home/blester/dev/work/baseline/python/baseline/pytorch/torchy.py", line 200, in pytorch_linear
    l = nn.Linear(in_sz, out_sz)
  File "/home/blester/anaconda3/envs/dl/lib/python3.6/site-packages/torch/nn/modules/linear.py", line 51, in __init__
    self.weight = Parameter(torch.Tensor(out_features, in_features))
TypeError: new() received an invalid combination of arguments - got (NoneType, int), but expected one of:
 * (torch.device device)
 * (torch.Storage storage)
 * (Tensor other)
 * (tuple of ints size, torch.device device)
      didn't match because some of the arguments have invalid types: (NoneType, int)
 * (object data, torch.device device)
      didn't match because some of the arguments have invalid types: (NoneType, int)
```

It seems this is missing a line that defaults `d_ff` like the `FFN` class used to

```python
d_ff = d_ff if d_ff is not None else 4 * d_model
```

Reply:
Yep, you're right. Fixed and base mead file running as expected now